sql/sql_connect.cc

```diff
@@ -826,6 +826,26 @@ bool init_new_connection_handler_thread()
   return 0;
 }
 
+static bool check_require_secured_transport(THD *thd)
+{
+  Vio *vio= thd->net.vio;
+  if (opt_require_secure_transport)
+  {
+    enum enum_vio_type type= vio_type(vio);
+
+    return
+#ifdef HAVE_OPENSSL
+      (type != VIO_TYPE_SSL) &&
+#endif
+#ifdef _WIN32
+      (type != VIO_TYPE_NAMEDPIPE);
+#else
+      (type != VIO_TYPE_SOCKET);
+#endif
```

Comment:
do you need all these #ifdef's ? they make the code difficult to read. Values are defined unconditionally, so it seems that you can remove ifdefs and the code will still compile file.

Reply:
Those `#ifdef`s predate this PR, as you can see from [the diff of `sql_acl.cc` above](https://github.com/MariaDB/server/pull/2581/files#diff-df544694418bef1c4bc6cdc5211ca133e7ad4d31901f16d0fdee8df6e4debe89). @Chupsy has simplified them considerably here.

> Values are defined unconditionally, …

Right. https://github.com/MariaDB/server/blob/HEAD/include/violite.h#L38-L45

> … so it seems that you can remove ifdefs and the code will still compile file.

There are a fairly enormous number of other places in the code which are wrapped in unnecessary `#ifdef HAVE_OPENSSL`, in that case :sweat_smile:

Reply:
I am not totally certain on how #ifdef work so I just optimized them with @dlenski, not sure what would happen if we removed them. Tell me if you want me to remove them completely from the code

Reply:
Based on what @vuvova said and what I see in the code, it seems like you could simply change this section as follows…

```cpp
    return
      (type != VIO_TYPE_SSL) &&
      (type != VIO_TYPE_NAMEDPIPE) &&
      (type != VIO_TYPE_SOCKET);
```

… and it should still compile and work correctly on all platforms, even if MariaDB is (somehow) built without TLS support, and even though named pipes are only used on Windows builds, and Unix sockets are only used on POSIX builds.